storagenode/piecestore/endpoint.go

```diff
@@ -265,6 +265,8 @@ func (endpoint *Endpoint) Upload(stream pb.DRPCPiecestore_UploadStream) (err err
 		uploadRate := float64(0)
 		if dt.Seconds() > 0 {
 			uploadRate = float64(uploadSize) / dt.Seconds()
+			mon.FloatVal("upload_rate_bytes_per_sec_meter").Observe(uploadRate)
```

Comment:
I'm not sure if we want to have another metric that we already have split into 3 different ones depending on whether was an error.

Reply:
I think that in both cases, there is no common metric for upload/download to indicate download rate/upload rate, which is may be the point of this ticket https://github.com/storj/storj/issues/3918

Adding this metric would provide a single consistent metric for speed (in whatever case whether upload/download passes for fails.

To be precise, in case of success, we would have `upload_success_rate_bytes_per_sec`, in case of failure `upload_failure_rate_bytes_per_sec` and in case of cancelled `upload_failure_rate_bytes_per_sec`. Thus a generic rate metric regardless of failure/success is needed.

However, if judged to be not necessary, then https://github.com/storj/storj/issues/3918 can be closed with a few comment to mention that the metrics are available already. 


Thougths ?

Reply:
#3918 doesn't say that much so I don't know if they referred to having that value in the web UI or if it's OK to get it from Monkit.

In the case that they were comfortable with Monkit, I'm not sure if they were aware of the existing metrics (perhaps they were added later), and if they were aware if the problem is that they want to have a unique metric as you added.

My comment is an observation rather than rejection or approval of your changes. I posted your PR internally for somebody of the team in charge of this part asses it properly, however, nobody has dropped by.

I'm going to repost it internally and follow up with it until somebody reviews it.

Reply:
Thanks.

In the beginning I intent to put the measurement on the dashboard. However, the issue is that information on Dashboard mostly comes from satellite. 

So to make it available on Dashboard, we would need to send aggregated speed every x hours to satellite. I guess that it is doable, but since it does not mention clearly in the ticket, I seriously wonder if it worth the effort spent or not.

Reply:
thanks for the reactivity

Reply:
Thanks for this PR @LaPetiteSouris.

I do share the same concern as @ifraixedes.

If I understand the ticket correctly, the point of the ticket is to have the upload and download rates on the storagenode dashboard like you said, and that's something I strongly agree to. If that's the case, then it makes sense to have a generic metric but we'd still need to display separately, the failure rate and success rate for uploads and downloads.